_posts/2015-06-25-infrastructure-as-code.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: post
+title: "Infrastructure as Code"
+date: "2015-06-25 09:00:00"
+icon: wrench
+tags: infrastructure, configuration management, continuous integration, jenkins
+categories: tech-stories
+authors: ["Jens Fischer", "Dirk Jablonski"]
+---
+
+Here at ePages we heavily rely on [Continuous Integration (CI)](http://www.martinfowler.com/articles/continuousIntegration.html) to make sure our code is always in a good shape. A mixed language stack mainly composed of `Perl`, `Java` and `JavaScript` spread over multiple teams brings in different build processes and toolings needed to produce the artifacts our software is composed of, e.g. `make`, `gradle` and `grunt`. [Jenkins](https://jenkins-ci.org/) is the software we use as our CI tool. For distributing the load created by each build process we leverage the [master/slave feature](https://wiki.jenkins-ci.org/display/JENKINS/Step+by+step+guide+to+set+up+master+and+slave+machines) provided by Jenkins. We do not run all build jobs on all Jenkins instances, but already spread out specific jobs to dedicated Jenkins instances. Nevertheless some teams still share their build infrastructure, thus lacking independance e.g. when it comes to introducing new Jenkins plugins or other tools needed for their build process.
+
+# Jenkins configuration
+
+Jenkins makes it very easy to introduce new functionality by installing one of the [many available plugins](https://wiki.jenkins-ci.org/display/JENKINS/Plugins). Unfortunately not every plugin turns out to work as expected or in the worst case can potentially bring down the whole Jenkins instance. It is a bad idea to evaluate unknown plugins by installing them into your production CI infrastructure. Even making changes to a job configuration without adding any new plugins can lead to broken build jobs, negatively affecting the productivity of whole teams.
```

Comment:
Rephrase the beginning of the sentence "It is a bad idea" - sounds very negative.